pkg/preparation/storacha/storacha.go

```diff
@@ -78,10 +101,32 @@ func (a API) SpaceBlobAddShardsForUpload(ctx context.Context, uploadID id.Upload
 			locationCommitment,
 		)
 		if err != nil {
-			return fmt.Errorf("failed to replicate shard %s: %w", shard.CID(), err)
+			return fmt.Errorf("failed to replicate shard %s: %w", shard.ID(), err)
+		}
+
+		// On shards too small to compute a CommP, just skip the `filecoin/offer`.
+		switch {
+		case shard.Size() < commp.MinPiecePayload:
+			log.Warnf("skipping `filecoin/offer` for shard %s: size %d is below minimum %d", shard.ID(), shard.Size(), commp.MinPiecePayload)
+		case shard.Size() > commp.MaxPiecePayload:
+			log.Warnf("skipping `filecoin/offer` for shard %s: size %d is above maximum %d", shard.ID(), shard.Size(), commp.MaxPiecePayload)
+		default:
+			pieceDigest, _, err := commpCalc.Digest()
+			if err != nil {
+				return fmt.Errorf("failed to get piece digest for shard %s: %w", shard.ID(), err)
+			}
+
+			shardPieceCID, err := commcid.DataCommitmentToPieceCidv2(pieceDigest, shard.Size())
+			if err != nil {
+				return fmt.Errorf("failed to get piece CID for shard %s: %w", shard.ID(), err)
+			}
+
+			_, err = a.Client.FilecoinOffer(ctx, spaceDID, cidlink.Link{Cid: shard.CID()}, cidlink.Link{Cid: shardPieceCID})
```

Comment:
❓ do we want to make any assertions on these receipts?

Reply:
I don't think so; that's not a receipt, it's a `filecoincap.OfferOk`. It only contains data we passed in ourselves. If the invocation fails, we'll get an error on `err`.